### DISCUSSION TITLE
Can't remove any widgets

### DISCUSSION THREAD

**Discussion by wisplite** [Textualize/textual#1494] [Q&A] [answered]
I'm attempting to remove widgets from the on_ready event using `self.query_one(MyWidget).remove()`. I can change reactive variables fine, but when I try to remove the same widget, I get a `NoActiveAppError`. Am I missing something?

**Comment by davep** [comment, marked as answer]
I've seen this particular error mentioned most often when people are using threads. Is that the case here? If so [this exchange over on the Discord server a while back](https://discord.com/channels/1026214085173461072/1033754296224841768/1053047210343932004) might be of interest.

**Reply by wisplite** [reply]
I'm not actually using threads, at least not actively. I'm trying to implement `python-socketio` into my app, and removing a widget upon a socket event is causing errors. `python-socketio` might implement threading in some way, though.

Edit: Oh wait, that's exactly what the first person had. I'll try and see if it's the same error.

Okay, it seems like I should try and implement the socketio `AsyncClient()` method, instead of the standard `Client()` method. I think the async one implements asyncio, while the standard one appears to do weird threading stuff behind the scenes.

**Comment by wisplite** [comment]
Took a bit to figure out, but switching to an `AsyncClient()` and defining it within the app class fixed it! Thanks for the help!

**Reply by davep** [reply]
Glad to hear you got it sorted.
